src/components/artsy.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from "react"
+import * as sharify from "sharify"
+
+/**
+ * The currently signed-in user.
+ */
+export interface CurrentUser extends sharify.CurrentUser {}
+
+/**
+ * The Artsy specific props injected by the higher-order component produced by `ContextConsumer`.
+ *
+ * @see {@link ContextProvider}
+ * @see {@link ContextConsumer}
+ *
+ * @todo Once this PR for `rest` https://github.com/Microsoft/TypeScript/pull/13470 lands we’ll be able to not make this
+ *       optional and simply remove it from the props that a component wrapped with the `ContextConsumer` HOC accepts.
+ */
+export interface ContextProps {
+  artsy?: Props
+}
+
+interface Props {
+  currentUser?: CurrentUser,
+}
+
+const ContextTypes = {
```

Comment:
We should use the [prop-types](https://www.npmjs.com/package/prop-types) package instead as React.PropTypes is deprecated 

> Note: As of React v15.5 the React.PropTypes helper is deprecated, and we recommend using the prop-types library to define contextTypes.